### DISCUSSION TITLE
how the heck do i run simple python code in phoenix?

### DISCUSSION THREAD

**Discussion by olitomar** [phcode-dev/phoenix#1111] [Q&A] [answered]
I can see the live preview for the html file, but i don't see a terminal or even the "run" button to run code and see the output. Help!

**Comment by davebritton** [comment]
Do you want the code to run on your own computer that is running the browser with phoenix, or on a remote webserver? If the latter, you need the webserver to be running Node.js. You need to launch a shell running within javascript:
Check out http://www.devdoc.net/web/developer.mozilla.org/en-US/docs/Web/JavaScript/Shells.html
Or, if you have remote ssh access to the server just run python there. 
Another approach is to set up a template engine like Jinja2 to run python in the web server.

**Reply by olitomar** [reply]
I think I understand less now. Could you dumb it down for me a bit, please?

**Reply by olitomar** [reply]
I'll rephrase my question: how do i make the green play button appear on phoenix when i have a python file open?

**Reply by davebritton** [reply, marked as answer]
Sorry, I misunderstood. I thought you wanted to run a python-driven website built using Phoenix.  You cannot directly run a python program that you have edited with Phoenix from inside the Phoenix editor. Try this instead: https://eric-ide.python-projects.org/
which is specifically for editing python projects.

**Reply by olitomar** [reply]
Thanks for your help. I will do that!
